lib/jormungandr/test/integration/Cardano/Wallet/Jormungandr/NetworkSpec.hs

```diff
@@ -0,0 +1,113 @@
+{-# LANGUAGE LambdaCase #-}
+module Cardano.Wallet.Jormungandr.NetworkSpec
+    ( spec
+    ) where
+
+import Prelude
+
+import Cardano.Launcher
+    ( Command (..), StdStream (..), launch )
+import Cardano.Wallet.Jormungandr.Api
+    ( BlockId (..) )
+import Cardano.Wallet.Jormungandr.Network
+    ( BaseUrl (..)
+    , Scheme (Http)
+    , defaultManagerSettings
+    , getBlock
+    , getTipId
+    , mkJormungandr
+    , newManager
+    , postSignedTx
+    )
+import Cardano.Wallet.Network
+    ( ErrNetworkTip (..), ErrNetworkUnreachable (..), ErrPostTx (..) )
+import Cardano.Wallet.Primitive.Types
+    ( Block (..), Hash (..), Tx (..) )
+import Control.Concurrent
+    ( threadDelay )
+import Control.Concurrent.Async
+    ( async, cancel )
+import Control.Monad.Trans.Except
+    ( runExceptT )
+import Test.Hspec
+    ( Spec
+    , afterAll
+    , beforeAll
+    , describe
+    , it
+    , shouldBe
+    , shouldReturn
+    , shouldSatisfy
+    , xit
+    )
+
+import qualified Data.ByteString as BS
+
+spec :: Spec
+spec = do
+    describe "Happy paths"
+        $ beforeAll startNode $ afterAll closeNode $ do
+        it "get tip (should be genesis)" $ \(_, client) -> do
+            (BlockId (Hash tipHash)) <- run $ getTipId client
+            tipHash `shouldBe` genesisHash
+            (BS.length tipHash) `shouldBe` 32
+
+        it "get block (genesis)" $ \(_, client) -> do
+            b <- run $ getBlock client (BlockId $ Hash genesisHash)
+            length (transactions b) `shouldBe` 1
+
+    describe "Error paths" $ beforeAll newClient $ do
+        it "gets a 'ErrNetworkUnreachable' if jormungandr isn't up (1)"
+            $ \bridge -> do
+            let msg x = "Expected a ErrNetworkUnreachable' failure but got "
+                    <> show x
+            let action = do
+                    res <- runExceptT $ getTipId bridge
+                    res `shouldSatisfy` \case
+                        Left (ErrNetworkTipNetworkUnreachable _) -> True
+                        _ -> error (msg res)
+            action `shouldReturn` ()
+
+        xit "gets a 'ErrNetworkUnreachable' if jormungandr isn't up (2)"
+            $ \bridge -> do
+            let msg x = "Expected a ErrNetworkUnreachable' failure but got "
+                    <> show x
+            let action = do
+                    res <- runExceptT $ postSignedTx bridge (txEmpty, [])
+                    res `shouldSatisfy` \case
+                        Left (ErrPostTxNetworkUnreachable (ErrNetworkUnreachable _)) ->
+                            True
+                        _ ->
+                            error (msg res)
+            action `shouldReturn` ()
+
+  where
+    txEmpty = Tx [] []
+
+    genesisHash = "&\198@\163\222\t\183C\152\193L\160\161\&7\236x\245\229\EOT\175\177\167\131\190\b\b/\174\212\177:\179"
+
+    run x = (either (error . show) id) <$> runExceptT x
```

Comment:
you probably want to use [`unsafeRunExceptT`](https://input-output-hk.github.io/cardano-wallet/haddock/edge/cardano-wallet-core-2019.5.24/Cardano-Wallet.html#v:unsafeRunExceptT). I do agree that this isn't the easiest helper function to locate in the codebase .... :|